### PR TITLE
update visual studio version mappings in windows platform

### LINKED ISSUE #28: If not specified, default compiler should match official ones
For [Python 2.7](https://docs.python.org/2.7/using/windows.html#compiling-python-on-windows) to [Python 3.2](https://docs.python.org/3.2/using/windows.html#compiling-python-on-windows): VS2008
For [Python 3.3](https://docs.python.org/3.3/using/windows.html#compiling-python-on-windows) to [Python 3.4](https://docs.python.org/3.4/using/windows.html#compiling-python-on-windows): VS2010
For [Python 3.5 and above](https://docs.python.org/3/using/windows.html#compiling-python-on-windows): VS2015

Following code should be tweaked: https://github.com/scikit-build/scikit-build/blob/45c087adf85c83c454744a246b9a4c5b99273d2f/skbuild/platform_specifics/windows.py#L14-L19

### PR DESCRIPTION
**Pull request by opadron**
fixes #28

### CODE REVIEW

**Comment by opadron** [comment]
@scikit-build/developers Ready for review

**Comment by thewtex** [comment]
:+1:

**Comment by jcfr** [comment]
Nice.

Do you think you could update the default test to check that the generator used by skbuild is the expected one ? 

See https://github.com/scikit-build/scikit-build/blob/9e2dabc054e903c571ed5d335afa45804041e215/appveyor.yml#L57-L69

File: tests/test_skbuild.py [hunk 1]
```diff
@@ -6,3 +6,53 @@
...
+            "Visual Studio 9 2008" if py_27_33 else
+            "Visual Studio 10 2010" if py_33_35 else
+            "Visual Studio 14 2015"
+        ) + (
+            "Win64" if arch == "x64" else
+            "ARM" if arch == "ARM" else
+            ""
+        )
+
+        assert(get_generator() == generator)
```
Comment (opadron) [inline tests/test_skbuild.py]: @jcfr what do you think about this?

File: skbuild/cmaker.py [hunk 2]
```diff
@@ -66,9 +66,19 @@ def configure(self, clargs=(), generator_id=None):
             The string representing the CMake generator to use.  If None, uses defaults for your platform.
         """
 
-        clargs, generator_id = pop_arg('-G', clargs)
-        generator_id = get_platform().get_best_generator(
-            generator_id)
+        # if no provided default generator_id, check environment
+        if generator_id is None:
+            generator_id = os.environ.get("CMAKE_GENERATOR")
```
Comment (jcfr) [inline skbuild/cmaker.py]: Good for now. We may have to revisit later based on best practice associated distutils and next generation of python build system discussed on the "distutils" mailing list.

File: tests/test_skbuild.py [hunk 3]
```diff
@@ -6,3 +6,53 @@
...
+import platform
+import sys
+
+from skbuild.platform_specifics import get_platform
+
+def test_generator_selection():
+    version = sys.version_info
+    env_generator = os.environ.get("CMAKE_GENERATOR")
+    this_platform = platform.system().lower()
+    get_generator = get_platform().get_best_generator
```
Comment (jcfr) [inline tests/test_skbuild.py]: get_best_generator()

File: tests/test_skbuild.py (hunk 1 above)
Comment (jcfr) [inline tests/test_skbuild.py]: :+1:

File: tests/test_skbuild.py (hunk 3 above)
Comment (opadron) [inline tests/test_skbuild.py]: Right, this is just an alias. Did you want me to keep the name?

**Comment by jcfr** [comment]
LGTM

**Comment by jcfr** [comment]
Nice work :8ball:

**Pull request merged**